client/src/components/GamePortalContainer/GamePortalContainer.jsx

```diff
@@ -2,15 +2,295 @@ import React, { Component } from 'react';
+import LoopIcon from '@material-ui/icons/Loop';
 
 class GamePortalContainer extends Component {
     constructor(props) {
         super(props);
-        this.state = {};
+        this.state = {
+            isLoaded: false,
+            isUserPlayerSet: false,
+            isClickTrack: true,
+            isLoopPlayerSet: false,
+            isAllUserPlayerSet: false
+        };
 
-        Tone.Transport.bpm.value = this.props.room.bpm;
+        // get bindings out of the way:
+        this.createMetronome = this.createMetronome.bind(this);
+        this.createClickTrack = this.createClickTrack.bind(this);
+        this.createLoopPlayer = this.createLoopPlayer.bind(this);
+        this.createAllUserPlayer = this.createAllUserPlayer.bind(this);
+
+        this.handleRecordLoop = this.handleRecordLoop.bind(this);
+        this.handleRecordOverLoop = this.handleRecordOverLoop.bind(this);
+        this.handlePlaybackMerged = this.handlePlaybackMerged.bind(this);
+        this.handleToggleClickTrack = this.handleToggleClickTrack.bind(this);
+
+        this.startMicrophonePermissions = this.startMicrophonePermissions.bind(this);
+        this.stopMicrophoneAccess = this.stopMicrophoneAccess.bind(this);
+        this.startRecording = this.startRecording.bind(this);
+        this.stopRecording = this.stopRecording.bind(this);
+        this.stopRecordingLoop = this.stopRecordingLoop.bind(this);
+        this.saveAudio = this.saveAudio.bind(this);
+        this.saveLoopedAudio = this.saveLoopedAudio.bind(this);
+        
+        // singleton settings    
+        Tone.context.latencyHint = 'interactive';
+        Tone.Transport.bpm.value = this.props.room.bpm; // does this update??
+        Tone.Buffer.on('load', 
+            () => {this.setState({isLoaded: true});}
+        );
+
+        // useful Tone.Time objects (DO THESE UPDATE?)
+        this.toneNumBars = Tone.Time(this.props.room.numBars, 'm');
+        this.toneTotalBars = Tone.Time(this.props.room.numBars * this.props.room.numLoops, 'm')
+    
+        // declare the players
+        this.loopPlayer = null;
+        this.userPlayers = null;
+        this.allUserPlayer = null;
+        this.metronome = null;
+        this.clickTrack = null;
+
+        // start and stop recording events
+        this.startRecordEvent = new Tone.Event(this.startRecording);
+        this.stopRecordEvent = new Tone.Event(this.stopRecording);
+        this.stopRecordLoopEvent = new Tone.Event(this.stopRecordingLoop);
+
+        if (!this.state.isLoopPlayerSet) this.createLoopPlayer();
+        if (!this.state.isAllUserPlayerSet) this.createAllUserPlayer();
+        this.createClickTrack();
+        this.createMetronome();
+
+    }
+    // -------------------------------------------------------------------------------------
+    // Tone Object Creation
+
+    createMetronome() {
+        let metronomeSynth = new Tone.MembraneSynth().toMaster();
+        metronomeSynth.volume.value = 5;
```

Comment:
maybe this should be controlled by the volume controller, so let the master volume set it?